### PR TITLE
Add fuzzing

### PR DESCRIPTION
**Pull request by Kixunil**
This adds fuzz targets using cargo-fuzz.

### CODE REVIEW

File: fuzz/fuzz_targets/rt.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#![no_main]
```
Comment (Kixunil) [inline fuzz/fuzz_targets/rt.rs]: Would like this file name to be more descriptive unless it's some kind of convention I don't know about.

**Review by Kixunil — COMMENTED** [review]
This looks good and would like to add it to CI. @RCasatta do you happen to know if there's an easy way to run this on top of MIRI? There's a bit of hopefully obvious `unsafe` code.

**Comment by Kixunil** [comment]
Thanks a lot!

**Comment by RCasatta** [comment]
I amended last commit for an error in the CI yml...

**Comment by Kixunil** [comment]
Ah, I think if you rebase it it will allow another PR?

**Comment by RCasatta** [comment]
#2

**Pull request merged**